openedx/core/djangoapps/content_libraries/collections/rest_api/v1/views.py

```diff
@@ -179,3 +182,38 @@ def destroy(self, request, *args, **kwargs):
         # TODO: Implement the deletion logic and emit event signal
 
         return Response(None, status=HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(detail=True, methods=['delete', 'patch'], url_path='contents', url_name='contents:update')
+    def update_contents(self, request, lib_key_str, pk=None):
+        """
+        Adds (PATCH) or removes (DELETE) Components to/from a Collection.
+
+        Collection and Components must all be part of the given library/learning package.
+        """
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+        library_obj = api.require_permission_for_library_key(
+            library_key,
+            request.user,
+            permissions.CAN_EDIT_THIS_CONTENT_LIBRARY,
+        )
+
+        serializer = ContentLibraryCollectionContentsUpdateSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        try:
+            count = api.update_collection_contents(
+                library_obj,
+                collection_pk=pk,
+                usage_keys=serializer.validated_data["usage_keys"],
+                remove=(request.method == "DELETE"),
+            )
+        except Collection.DoesNotExist as exc:
+            raise Http404() from exc
+
+        except Component.DoesNotExist as exc:
+            # Only allows adding/removing components that are in the collection's learning package.
+            raise ValidationError({
+                "usage_keys": _("Component(s) not found in library"),
+            }) from exc
```

Comment:
Copying from the other rest api PR:

We have a [`convert_exceptions`](https://github.com/openedx/edx-platform/blob/ea596d6a25e6608305be535e933146af015b4ec5/openedx/core/djangoapps/content_libraries/views.py#L129-L154) decorator in the `content_library` that we can move to a utils file and use in our functions here, so we don't need to re-raise our exceptions.

Reply:
Oh brilliant! I'll use that.